src/app/pages/task-calendar/scrub/scrub-form/scrub-form.component.ts

```diff
@@ -17,78 +18,105 @@ import helptext from '../../../../helptext/task-calendar/scrub/scrub-form';
-      tooltip: helptext.scrub_picker_tooltip,
-      required: true,
-    }, {
-      type: 'checkbox',
-      name: 'scrub_enabled',
-      placeholder: helptext.scrub_enabled_placeholder,
-      tooltip : helptext.scrub_enabled_tooltip,
-      value: true,
-    }
-  ];
+  public fieldConfig: FieldConfig[] = [];
+  public fieldSets: FieldSet[] = [
+    {
+      name:helptext.scrub_fieldsets[0],
+      class:'add-scrub',
+      label:true,
+      width:'300px',
+      config: [
+        {
+          type: 'select',
+          name: 'pool',
+          placeholder: helptext.scrub_volume_placeholder,
+          tooltip : helptext.scrub_volume_tooltip,
+          options: [],
+          required: true,
+          validation : helptext.scrub_volume_validation,
+          value: '',
+        }, {
+          type: 'input',
+          inputType: 'number',
+          name: 'threshold',
+          placeholder: helptext.scrub_threshold_placeholder,
+          tooltip: helptext.scrub_threshold_tooltip,
+          value: 35,
+          min: 0,
+          required: true,
+          validation: helptext.scrub_threshold_validation
+        }, {
+          type: 'input',
+          name: 'description',
+          placeholder: helptext.scrub_description_placeholder,
+          tooltip : helptext.scrub_description_tooltip,
+        }, {
+          type: 'scheduler',
+          name: 'scrub_picker',
+          placeholder: helptext.scrub_picker_placeholder,
+          tooltip: helptext.scrub_picker_tooltip,
+          required: true,
+        }, {
+          type: 'checkbox',
+          name: 'enabled',
+          placeholder: helptext.scrub_enabled_placeholder,
+          tooltip : helptext.scrub_enabled_tooltip,
+          value: true,
+        }
+      ]
+    },
+    {
+      name:'divider',
+      divider:true
+    },
+  ]
 
   protected volume_field: any;
   protected month_field: any;
   protected day_field: any;
-  protected mintue_field: any;
+  protected minute_field: any;
   protected hour_field: any;
   protected daymonth_field: any;
 
-  constructor(protected router: Router, protected taskService: TaskService, protected userService: UserService, protected entityFormService: EntityFormService) {
+  constructor(protected router: Router, protected taskService: TaskService, protected userService: UserService, protected entityFormService: EntityFormService,protected aroute: ActivatedRoute,) {
   }
 
-  preInit() {
-    this.volume_field = _.find(this.fieldConfig, { 'name': 'scrub_volume' });
+  preInit(entityForm) {
+    this.entityForm = entityForm;
+
+    this.aroute.params.subscribe(params => {
+      if (params.pk) {
+        this.pk = params.pk;
+      }
+    });
+
+    this.volume_field = _.find(this.fieldConfig, { 'name': 'pool' });
     this.taskService.getVolumeList().subscribe((res) => {
       for (let i in res.data) {
         this.volume_field.options.push({ label: res.data[i].vol_name, value: res.data[i].id });
```

Comment:
I'm getting errors about volume_field being undefined at this line, it also isn't populating the pool list